patroni/dcs/kubernetes.py

```diff
@@ -109,6 +116,12 @@ def server(self):
 
     @property
     def headers(self):
+        if self._token_expires_at and self._token_expires_at <= datetime.datetime.now():
+            try:
+                token = self._read_token_file()
+                self._headers['authorization'] = 'Bearer ' + token
```

Comment:
The same line is used twice, here and https://github.com/zalando/patroni/pull/2287/files#diff-0b47c783038695efd3a30f9797eb9699abc996d28c5ec9f040b81e63e995b068R57
Let's extract it to a separate method with the signature `def _set_token(self, token)`?